server/text_generation_server/utils/chunks.py

```diff
@@ -0,0 +1,20 @@
+from typing import Iterable
+
+from loguru import logger
+
+from text_generation_server.pb import generate_pb2
+
+
+def concat_text_chunks(chunks: Iterable[generate_pb2.InputChunk]) -> str:
```

Comment:
Is this method only to be future proof or is there a way today to have multiple text chunks?

Reply:
AFAIK we can currently only have multiple text chunks in a VLM models, so this was indeed only to future proof.

Reply:
Then mb we should take [0] and crash if len > 1 with an unreachable?

Reply:
Oh, we do need to iterate over chunks, because we are sending image chunks unconditionally during warmup, even for text-only models:

https://github.com/huggingface/text-generation-inference/blob/757223b352896a9b2b9df46c95c0afcaa0cdf9d4/router/client/src/v3/client.rs#L140

The current approach seems more robust? What do you think about logging a warning when `len(texts) > 1`?

Reply:
Updated to:

* Fail when there is more than one text chunk.
* Fail when there is no text chunk.
* Log at debug-level when there is a non-text chunk (only log because e.g. warmup sends an image chunk).